README.md

```diff
@@ -100,109 +100,141 @@ module.exports = i18n
 
-```js
-{
-  lexers: {
+  "defaultNamespace": "translation",
+  // Default namespace used in your i18next config         
+
+  "defaultValue": "",
+  // Default value to give to empty keys                   
+
+  "extension": ".json",
+  // Extension of the catalogs 
+  // Supports $LOCALE and $NAMESPACE injection
+
+  "filename": "$NAMESPACE",
+  // Filename of the catalogs                              
+  // Supports $LOCALE and $NAMESPACE injection
+
+  "indentation": 2,
+  // Indentation of the catalog files                      
+
+  "keepRemoved": false,
+  // Keep keys from the catalog that are no longer in code 
+
+  "keySeparator": ".",
+  // Key separator used in your translation keys
+  // If you want to use plain english keys, separators such as `.` and `:` will conflict. You might want to set `keySeparator: false` and `namespaceSeparator: false`. That way, `t('Status: Loading...')` will not think that there are a namespace and three separator dots for instance.        
+
+  // see below for more details
+  "lexers": {
     hbs: ['HandlebarsLexer'],
     handlebars: ['HandlebarsLexer'],
 
     htm: ['HTMLLexer'],
     html: ['HTMLLexer'],
 
-    js: ['JavascriptLexer'],
-    jsx: ['JavascriptLexer', 'JsxLexer'],
+    js: ['JavascriptLexer'], // if you're writing jsx inside .js files, change this to JsxLexer
+    jsx: ['JsxLexer'],
     mjs: ['JavascriptLexer'],
 
     default: ['JavascriptLexer']
-  }
-}
-```
+  },
 
-Note the presence of a `default` which will catch any extension that is not listed. There are 3 lexers available: `HandlebarsLexer`, `HTMLLexer` and `JavascriptLexer`. Each has configurations of its own. If you need to change the defaults, you can do it like so:
+  "lineEnding": auto,
+  // Control the line ending. See options at https://github.com/ryanve/eol
 
-```js
-{
-  lexers: {
-    hbs: [
-      {
-        lexer: 'HandlebarsLexer',
-        functions: ['translate', '__']
-      }
-    ],
-    js: [
-      {
-        lexer: 'JavascriptLexer',
-        acorn: {
-          plugins: {
-            jsx: true,
-            objectRestSpread: true,
-            es7: true
-          }
-        }
-      }
-    ]
-    // ...
-  }
-}
-```
+  "locales": ['en', 'fr'],
+  // An array of the locales in your applications          
 
-**`HandlebarsLexer` options**
+  "namespaceSeparator": ":",
+  // Namespace separator used in your translation keys   
+  // If you want to use plain english keys, separators such as `.` and `:` will conflict. You might want to set `keySeparator: false` and `namespaceSeparator: false`. That way, `t('Status: Loading...')` will not think that there are a namespace and three separator dots for instance.
 
-Option        | Description                 | Default
-------------- | --------------------------- | -------
-**functions** | Array of functions to match | `['t']`
+  "output": "locales",
+  // Where to write the locale files relative to the base
```

Comment:
For reference (and because I didn't realize until I dug into this) files are output following this pattern: `outputs[n]` + `$NAMESPACE` + `filename` + `extension`.

1. Add a `filepath` config option. (e.g.`$NAMESPACE`)
2. Change `filename` to include the namespace. (e.g. `$NAMESPACE/$LOCALE` )
3. Combine all sections into a single option and throw out the others. (e.g. `locales/$NAMESPACE/$LOCALE.json` )
4. Allow defining of full path in `outputs` (like opt 3) and ignore filename and extension if it's specified in outputs. 

Opt 1 and opt 4 are the most backwards compatible ones.  
Opt 1 makes the config more confusing (what's the difference between `outputs` and `filepath`?). 
Opt 3 reduces the number of options and requires explicitness which are generally good things... but what if the user just wants to customize extension OR filename? (Though I suspect this is a rare case.)  
Opt 4 is less appealing than opt 3 b/c bi-modal is a pain in the ass. (It'll probably be fairly easy here, but still. It's the principle of the thing.)
Opt 2 doesn't appeal at all. It seems like the least-good solution: it changes current behaviour AND it doesn't have the benefit of reducing variables or increasing explicitness.

I'd be most happy with Opt 3 or Opt 4. Thoughts?

Reply:
Opt 3 is a clear winner for me. We could also support the extension there (.json vs .yml)`.

And I don't care for backward compatibility, we are still in beta. Things can break.